uwdata/lyra
# Recompute layout if the band size of new ordinal scales is too small.

**Description of the problem this pull request fixes**
Spatial scales produced during data binding operations set the range to `width` or `height` as appropriate. With ordinal scales, however, this can produced a cramped layout if the domain has high cardinality. This PR recalculates the layout for new ordinal scales based on the minimum default band size from Vega-Lite. 

Changes proposed in this pull request:
- A new `computeLayout` function that is run for new scales. 

**This PR includes:**

**Steps to functionally test this:**
- Follow #331, particular testing binding nominal/ordinal fields on both x/y for multiple mark types and with low/high cardinality. Good data field candidates are Gapminder's `country` for high cardinality, and `year` (after switching it to a nominal type) for low. 
- Also test non-ordinal data binding to ensure layout is not recomputed.
